scheme/arm-cca/platform.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package arm_cca
+
+import (
+	"fmt"
+
+	"github.com/veraison/ear"
+	"github.com/veraison/psatoken"
+	"github.com/veraison/services/handler"
+	"github.com/veraison/services/log"
+	"github.com/veraison/services/scheme/common"
+	"github.com/veraison/services/scheme/common/arm"
+)
+
+type CCAPlatformAttester struct {
+}
+
+func (CCAPlatformAttester) PerformAppraisal(
+	appraisal *ear.Appraisal,
+	claimsMap map[string]interface{},
+	endorsements []handler.Endorsement,
+) error {
+	claims, err := common.MapToClaims(claimsMap)
+	if err != nil {
+		return fmt.Errorf("unable to get platform claims from platform claims map: %w", err)
+	}
+
+	// once the signature on the token is verified, we can claim the HW is
+	// authentic
+	appraisal.TrustVector.Hardware = ear.GenuineHardwareClaim
+
+	rawLifeCycle, err := claims.GetSecurityLifeCycle()
+	if err != nil {
+		return handler.BadEvidence(err)
+	}
+
+	lifeCycle := psatoken.CcaLifeCycleToState(rawLifeCycle)
+	if lifeCycle == psatoken.CcaStateSecured ||
+		lifeCycle == psatoken.CcaStateNonCcaPlatformDebug {
+		appraisal.TrustVector.InstanceIdentity = ear.TrustworthyInstanceClaim
+		appraisal.TrustVector.RuntimeOpaque = ear.ApprovedRuntimeClaim
+		appraisal.TrustVector.StorageOpaque = ear.HwKeysEncryptedSecretsClaim
+	} else {
+		appraisal.TrustVector.InstanceIdentity = ear.UntrustworthyInstanceClaim
+		appraisal.TrustVector.RuntimeOpaque = ear.VisibleMemoryRuntimeClaim
+		appraisal.TrustVector.StorageOpaque = ear.UnencryptedSecretsClaim
+	}
+
+	swComps := arm.FilterRefVal(endorsements, "sw-component")
+	match := arm.MatchSoftware(SchemeName, claims, swComps)
+	if match {
+		appraisal.TrustVector.Executables = ear.ApprovedRuntimeClaim
+		log.Debug("matchSoftware Success from Line 55")
+
+	} else {
+		appraisal.TrustVector.Executables = ear.UnrecognizedRuntimeClaim
+		log.Debug("matchSoftware Failed from Line 59")
+	}
+
+	platformConfig := arm.FilterRefVal(endorsements, "platform-config")
+	match = arm.MatchPlatformConfig(SchemeName, claims, platformConfig)
+
+	if match {
+		appraisal.TrustVector.Configuration = ear.ApprovedConfigClaim
+		log.Debug("matchPlatformConfig Success")
+
+	} else {
+		appraisal.TrustVector.Configuration = ear.UnsafeConfigClaim
+		log.Debug("matchPlatformConfig Failed")
+	}
+
+	appraisal.UpdateStatusFromTrustVector()
+
+	appraisal.VeraisonAnnotatedEvidence = &claimsMap
+
+	return nil
+}
```

Comment:
Related to https://github.com/veraison/services/pull/233/files#r1644911152 I suggest we make `PerformAppraisal` return the appraisal object:

```suggestion
func (CCAPlatformAttester) PerformAppraisal(
	claimsMap map[string]interface{},
	endorsements []handler.Endorsement,
) (*ear.Appraisal, error) {
	claims, err := common.MapToClaims(claimsMap)
	if err != nil {
		return nil, fmt.Errorf("unable to get platform claims from platform claims map: %w", err)
	}

    trustVector := ear.TrustVector{}
	// once the signature on the token is verified, we can claim the HW is
	// authentic
	trustVector.Hardware = ear.GenuineHardwareClaim

	rawLifeCycle, err := claims.GetSecurityLifeCycle()
	if err != nil {
		return nil, handler.BadEvidence(err)
	}

	lifeCycle := psatoken.CcaLifeCycleToState(rawLifeCycle)
	if lifeCycle == psatoken.CcaStateSecured ||
		lifeCycle == psatoken.CcaStateNonCcaPlatformDebug {
		trustVector.InstanceIdentity = ear.TrustworthyInstanceClaim
		trustVector.RuntimeOpaque = ear.ApprovedRuntimeClaim
		trustVector.StorageOpaque = ear.HwKeysEncryptedSecretsClaim
	} else {
		trustVector.InstanceIdentity = ear.UntrustworthyInstanceClaim
		trustVector.RuntimeOpaque = ear.VisibleMemoryRuntimeClaim
		trustVector.StorageOpaque = ear.UnencryptedSecretsClaim
	}

	swComps := arm.FilterRefVal(endorsements, "sw-component")
	match := arm.MatchSoftware(SchemeName, claims, swComps)
	if match {
		trustVector.Executables = ear.ApprovedRuntimeClaim
	} else {
		appraisal.TrustVector.Executables = ear.UnrecognizedRuntimeClaim
	}

	platformConfig := arm.FilterRefVal(endorsements, "platform-config")
	match = arm.MatchPlatformConfig(SchemeName, claims, platformConfig)

	if match {
		trustVector.Configuration = ear.ApprovedConfigClaim
	} else {
		appraisal.TrustVector.Configuration = ear.UnsafeConfigClaim
	}

    appraisal := ear.Appraisal{
        TrustVector: trustVector,
    }

	appraisal.UpdateStatusFromTrustVector()
	appraisal.VeraisonAnnotatedEvidence = &claimsMap

	return &appraisal, nil
}
```